libkbfs/quota_usage.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/kbfs/kbfsblock"
+)
+
+//ECQUCtxTagKey is the type for unique ECQU background opertaion IDs.
+type ECQUCtxTagKey struct{}
+
+// ECQUID == "ECQU"
+const ECQUID = "ECQU"
+
+type cachedQuotaUsage struct {
+	timestamp  time.Time
+	usageBytes int64
+	limitBytes int64
+}
+
+// EventuallyConsistentQuotaUsage keeps tracks of quota usage, in a way user of
+// which can choose to accept stale data to reduce calls into block servers.
+type EventuallyConsistentQuotaUsage struct {
+	config Config
+	log    logger.Logger
+
+	backgroundInProcess int32
+
+	mu     sync.RWMutex
+	cached cachedQuotaUsage
+}
+
+// NewEventuallyConsistentQuotaUsage creates a new
+// EventuallyConsistentQuotaUsage object.
+func NewEventuallyConsistentQuotaUsage(
+	config Config) *EventuallyConsistentQuotaUsage {
+	return &EventuallyConsistentQuotaUsage{
+		config: config,
+		log:    config.MakeLogger(ECQUID),
```

Comment:
It's more useful to have an instance-specific logger tag, rather than a struct-specific one, since we can already tell that it comes from the struct using the file name.  I would either get rid of the tag, or pass one in so we can distinguish between libfuse and KBFSOps requests.